packages/vhd-lib/src/Vhd/VhdSynthetic.js

```diff
@@ -0,0 +1,86 @@
+import { asyncMap } from '@xen-orchestra/async-map'
+import { VhdAbstract } from './VhdAbstract'
+import { FOOTER_SIZE, HEADER_SIZE, SECTOR_SIZE } from '../_constants'
+import { sectorsRoundUpNoZero, sectorsToBytes } from './_utils'
+
+import assert from 'assert'
+
+export class VhdSynthetic extends VhdAbstract {
+  #vhds = []
+  set header(_) {
+    throw new Error('Header is read only for VhdSynthetic')
+  }
+
+  get header() {
+    // this the VHD we want to synthetize
+    const vhd = this.#vhds[0]
+
+    // this is the root VHD
+    const rootVhd = this.#vhds[this.#vhds.length - 1]
+
+    // data of our synthetic VHD
+    // TODO: set parentLocatorEntry-s in header
+    return {
+      ...vhd.header,
+      tableOffset: FOOTER_SIZE + HEADER_SIZE,
+      parentTimestamp: rootVhd.header.parentTimestamp,
+      parentUnicodeName: rootVhd.header.parentUnicodeName,
+      parentUuid: rootVhd.header.parentUuid,
+    }
+  }
+
+  set footer(_) {
+    throw new Error('Footer is read only for VhdSynthetic')
+  }
+
+  get footer() {
+    // this is the root VHD
+    const rootVhd = this.#vhds[this.#vhds.length - 1]
+    return {
+      ...this.#vhds[0].footer,
+      dataOffset: FOOTER_SIZE,
+      diskType: rootVhd.footer.diskType,
+    }
+  }
+
+  static async open(vhds) {
+    const vhd = new VhdSynthetic(vhds)
+    return {
+      dispose: () => {},
+      value: vhd,
+    }
+  }
+
+  constructor(vhds) {
+    assert(vhds.length > 0)
+    super()
+    this.#vhds = vhds
+    this.sectorsPerBlock = vhds[0].header.blockSize / SECTOR_SIZE
+    this.sectorsOfBitmap = sectorsRoundUpNoZero(this.sectorsPerBlock >> 3)
+    this.fullBlockSize = sectorsToBytes(this.sectorsOfBitmap + this.sectorsPerBlock)
+    this.bitmapSize = sectorsToBytes(this.sectorsOfBitmap)
```

Comment:
Why is it necessary?

Wouldn't it be better to do `this.header = vhds[0].header`?

Reply:
Ok, I see that `header` is read-only for this type of VHD.

But it makes me wonder if we handled these entries correctly: instead of computing them statically when setting the header, it may be best to create getters for them, what do you think?

Reply:
they can only be computed when header blockSize is set, and are heavily utilized through the code. I think precomputing them is a good way if we don't want to add the check overhead, and they only cost us 4 integers in memory per Vhd

Reply:
I don't mind the overhead, I prefer it to the having the logic duplicated.